[ADD] field_service: create new page about worksheet (#9795)
odoo/documentation: content/applications/services/field_service/worksheets.rst

.. warning::
   Worksheet templates are designed using **Studio** and clicking :guilabel:`Design Template`
   automatically installs the **Studio** app, which impacts your price plan.

Create a worksheet template
---------------------------

To create your **worksheet templates**, go to :menuselection:`Field Service --> Configuration -->
Worksheet Templates`. Click :guilabel:`New` and give your worksheet template a name. You can also
add a company and a color to add clarity if needed and manually :guilabel:`Save`. Then click
:guilabel:`Design Template` to open **Studio** and customize your worksheet template.

In Studio click :guilabel:`Add`. Drag and drop the items that you want from the left column
into your worksheet. To rearrange the items, just drag and drop them in the order that you want.
Click on the items you added to your worksheet to customize their properties.

When your Worksheet Template is complete, click :guilabel:`Close` in the top right corner of the
page to leave **Studio**.

.. seealso::
   - :doc:`Fields and widgets in Studio <../../studio/fields>`

Add a worksheet template to a Field Service task
================================================

> Go to your field service task and select a :guilabel:`Worksheet Template` and click
> :guilabel:`Save`.

Reviewer: ```suggestion
Go to your field service task, select a :guilabel:`Worksheet Template`, and click :guilabel:`Save`.
```

> By default, the :guilabel:`Default worksheet` template is selected. To define another default

Reviewer: ```suggestion
By default, the :guilabel:`Default Worksheet` template is selected. To define another default
```

> worksheet template, click the :guilabel:`➔` (:guilabel:`Internal link`) icon that appears when you

Reviewer: I am pretty sure it was an arrow before but now I see this instead (this means we would need to change the screenshot as well ...)
![image](https://github.com/odoo/documentation/assets/128469033/10373c4a-b7e7-4c50-a8fb-3ca69172c692)

===================
Worksheet Templates
===================

**Worksheets** help your field service workers perform and report the tasks they do on site.
They can feature various information, such as instructions, to-do lists, etc. You can
also format your worksheet using checkboxes, bullet points, blank fields to fill in, HTML, and add
files, images, links, and more.

Because it is common for businesses to have their workers perform the same type of field service
repeatedly, Odoo offers the option to create custom **Worksheet Templates**. By doing so, you
don’t have to create the same worksheet every time you plan the same type of field service task.

Configuration
=============

To use **worksheets** in **Field Service**, the **Worksheet Template** feature must be enabled

> first. Go to :menuselection:`Field Service --> Configuration --> Settings --> Worksheet Templates`

Author: Yes that's what I meant (question 2), your suggestion makes it more clear :)

To create your **worksheet templates**, go to :menuselection:`Field Service --> Configuration -->
Worksheet Templates`. Click :guilabel:`New`, give your worksheet template a name, then click
:guilabel:`Design Template`. Doing so will open the **Studio**.

.. important::
   Clicking :guilabel:`Design Template` will automatically install the **Studio** app in your
   database, which is required to design worksheet templates. Adding more apps to your database may
   impact your price plan.

In the Studio click :guilabel:`Add`. Drag and drop the items that you want from the left column
into your worksheet. To rearrange the items, just drag and drop them in the order that you want.
Click on the items you added to your worksheet to customize their properties.

When your Worksheet Template is complete, click :guilabel:`Close` in the top right corner of the
page to leave **Studio**.

Add a Worksheet Template to a Field Service task
================================================

Go to your Field Service task and select a :guilabel:`Worksheet Template`. Remember that the
:guilabel:`Task Title` and :guilabel:`Customer` fields are mandatory. Click :guilabel:`Save`.

> Assign a default Worksheet Template
> ===================================
> 
> If you have configured multiple Worksheet Templates, you can select one to be set up by
> default when you create a new Field Service task. Go to your task, drag your mouse over the name of
> your project to make the green arrow and the **internal link** tab appear, click on it.

Reviewer: I don't think the title is necessary here and it looks a little odd with just 1 line udner the previous one. Instead, I would maybe do something like this to 1) introduce the concept of a default template (which was done in the title) and 2) be a little more direct:

```suggestion
By default, the :guilabel:`Default worksheet` template is selected. To define another default worksheet template,  
click the :icon:`oi-arrow-right` (right-arrow) button that appears when you hover your mouse over the :guilabel:`Project` field on the task form. 

```
see this doc for how to add icons directly into the text like I did in my suggestion https://www.odoo.com/documentation/master/contributing/documentation/rst_cheat_sheet.html#icons

Author: Thank you for this tip!

> .. image:: worksheets/template-default.png

Reviewer: I would crop this screenshot to only show the left column (which will highlight what you really want to show.

Author: Done, including png quant :)

Reviewer: I think something went wrong somewhere because from what I can see the image is not compressed:
![image](https://github.com/odoo/documentation/assets/128469033/d42d1dbe-7dea-4571-8e4a-90823a7a428b)
I also think we could get rid of the blank part

   :scale: 75%
   :alt: Mouse on project name to setup default worksheet template

> Then click
> on the :guilabel:`Settings` tab. Under **Field Service**, make sure the :guilabel:`Worksheet`
> checkbox is ticked and select the :guilabel:`Worksheet Template` you want to set up as default.
> :guilabel:`Save` your changes.

Reviewer: ```suggestion
Then, in the :guilabel:`Settings` tab,  scroll down to the :guilabel:`Field service` section and select the :guilabel:`Worksheet Template` you want to set up as default.
```
- make sure the Worksheet checkbox is ticked: since we enabled it earlier, i don't think we need to do it again?
- Field service = guilabel and not bold since it's a UI element
- no need to really manually save so i remvoed that as well

Use worksheets on site
======================

> Field service workers can complete the worksheet to document the work they are doing on site. Go to
> your :guilabel:`Task` and click the :guilabel:`Worksheet` Smart Button. Fill the blank fields
> and :guilabel:`Save`.

Reviewer: ```suggestion
To complete the worksheet on site, access the task and click the :guilabel:`Worksheet` smart button.
```
- I suggest removing the last part because what's in the worksheet can vary a lot and are not just "blanl fields", if that makes sense? (e.g., you can also have check boxes, or dropdown lists, or images, etc.) and i think it's kind of implied in the "complete the worksheet" part at the beginning?
- Saving is done automatically here (manual saving is optional so no need to mention it I think)

.. note::

>    - As soon as you save a worksheet, the appearance of the **Worksheet** smart button on
>      your task will change and display **Worksheet Complete** instead, regardless some fields are
>      left blank.
>    - Any field set up as **required** has to be filled for a worksheet to be saved.

Reviewer: ```suggestion
   - As soon as you save a worksheet, the appearance of the :guilabel:`Worksheet` smart button on
     the task changes to :guilabel:`Worksheet Complete` instead, even if some fields are
     left blank.
   - Any field defined as :guilabel:`Required` has to be filled for a worksheet to be saved.
```
- these are all UI elements and should therefore be guilabel (+ Required with a capital R)
- regardless needs "of" afterwards so the sentence would be "regardless of some fields being left blank", which is very heavy IMO
